tests/langchain/chain.py

```diff
@@ -0,0 +1,80 @@
+import mlflow
+import os
+from typing import Any, List, Optional
+
+from langchain.document_loaders import TextLoader
+from langchain.embeddings.fake import FakeEmbeddings
+from langchain.prompts import ChatPromptTemplate
+from langchain.schema.output_parser import StrOutputParser
+from langchain.schema.runnable import RunnablePassthrough
+from langchain.text_splitter import CharacterTextSplitter
+from langchain.vectorstores import FAISS
+from mlflow.langchain._rag_utils import _set_chain
+
+
+def get_fake_chat_model(endpoint="fake-endpoint"):
+    from langchain.callbacks.manager import CallbackManagerForLLMRun
+    from langchain.chat_models import ChatDatabricks, ChatMlflow
+    from langchain.schema.messages import BaseMessage
+    from langchain_core.outputs import ChatResult
+
+    class FakeChatModel(ChatDatabricks):
+        """Fake Chat Model wrapper for testing purposes."""
+
+        endpoint: str = "fake-endpoint"
+
+        def _generate(
+            self,
+            messages: List[BaseMessage],
+            stop: Optional[List[str]] = None,
+            run_manager: Optional[CallbackManagerForLLMRun] = None,
+            **kwargs: Any,
+        ) -> ChatResult:
+            response = {
+                "choices": [
+                    {
+                        "index": 0,
+                        "message": {
+                            "role": "assistant",
+                            "content": "Databricks",
+                        },
+                        "finish_reason": None,
+                    }
+                ],
+            }
+            return ChatMlflow._create_chat_result(response)
+
+        @property
+        def _llm_type(self) -> str:
+            return "fake chat model"
+
+    return FakeChatModel(endpoint=endpoint)
+
+
+path = "tests/langchain/state_of_the_union.txt"
+text_path = mlflow.langchain._rag_utils.__databricks_rag_config_path__ or path
```

Comment:
This file is always going to exist on the local filesystem, so if `__databricks_rag_config_path__` is `None`, this test won't catch it. Can we:

1. Create a new file in a temporary directory and have this script reference it by environment variable, e.g.

```
text_path = os.environ[...] or mlflow.langchain._rag_utils.__databricks_rag_config_path__
```

3. After the model packaging step, delete the file before the model is loaded / served

Reply:
I completely removed setting the path `state_of_the_union.txt` to path since it should always use `mlflow.langchain._rag_utils.__databricks_rag_config_path__`. So now we can make sure it will always exists.